subprojects/language-native/src/main/java/org/gradle/language/nativeplatform/internal/incremental/DefaultSourceIncludesSearchPath.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.language.nativeplatform.internal.incremental;
+
+import com.google.common.base.Objects;
+import org.gradle.api.internal.changedetection.state.FileSnapshot;
+import org.gradle.api.internal.changedetection.state.FileSystemSnapshotter;
+import org.gradle.internal.file.FileType;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class DefaultSourceIncludesSearchPath implements SourceIncludesSearchPath {
+    private final List<File> includePaths;
+    private final FileSystemSnapshotter fileSystemSnapshotter;
+    private final Map<File, Map<String, IncludeFileImpl>> includeRoots;
+
+    public DefaultSourceIncludesSearchPath(List<File> includePaths, FileSystemSnapshotter fileSystemSnapshotter) {
+        this.includePaths = includePaths;
+        this.fileSystemSnapshotter = fileSystemSnapshotter;
+        this.includeRoots = new HashMap<File, Map<String, IncludeFileImpl>>();
+    }
+
+    @Override
+    public SourceIncludesSearchPath asQuotedSearchPath(File sourceFile) {
```

Comment:
I would just make this simpler and pass the extra information to searchForDependency vs building a specialized anonymous class here.